datastream-server/src/main/java/com/linkedin/datastream/server/CoordinatorEventBlockingQueue.java

```diff
@@ -5,47 +5,101 @@
  */
 package com.linkedin.datastream.server;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.HashSet;
+import java.util.List;
 import java.util.Queue;
 import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.LinkedBlockingQueue;
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import com.codahale.metrics.Counter;
+import com.codahale.metrics.Gauge;
+import com.codahale.metrics.MetricRegistry;
+
+import com.linkedin.datastream.metrics.BrooklinCounterInfo;
+import com.linkedin.datastream.metrics.BrooklinGaugeInfo;
+import com.linkedin.datastream.metrics.BrooklinMetricInfo;
+import com.linkedin.datastream.metrics.DynamicMetricsManager;
+import com.linkedin.datastream.metrics.MetricsAware;
+
 
 /**
- * A blocking queue for {@link Coordinator} events
+ * A blocking queue for {@link Coordinator} events. Includes two metrics, a
+ * {@link Gauge} and {@link Counter}. The gauge provides the queue size and
+ * the counter increments when duplicate events are {@code put()}.
+ *
  * @see CoordinatorEvent.EventType
  */
-public class CoordinatorEventBlockingQueue {
+public class CoordinatorEventBlockingQueue implements MetricsAware {
 
   private static final Logger LOG = LoggerFactory.getLogger(CoordinatorEventBlockingQueue.class.getName());
+  private static final Set<BrooklinMetricInfo> METRIC_INFOS = ConcurrentHashMap.newKeySet();
+
+  static final String COUNTER_KEY = "duplicateEvents";
```

Comment:
I don't see a lot of practical value in measuring duplicateEvents with a counter.

Reply:
Let's leave it. Having an extra metric won't hurt.